File: services/src/main/java/com/liferay/mobile/sdk/v7/microblogsentry/MicroblogsEntryService.java
package com.liferay.mobile.sdk.v7.microblogsentry;

import com.liferay.mobile.sdk.Call;
import com.liferay.mobile.sdk.annotation.Param;
import com.liferay.mobile.sdk.annotation.Path;
import java.lang.Integer;
import java.lang.String;
import org.json.JSONArray;
import org.json.JSONObject;

@Path("/microblogs.microblogsentry")
public interface MicroblogsEntryService {
  @Path("/add-microblogs-entry")
  Call<JSONObject> addMicroblogsEntry(@Param(name = "userId") long userId, @Param(name = "content") String content, @Param(name = "type") int type, @Param(name = "parentMicroblogsEntryId") long parentMicroblogsEntryId, @Param(name = "socialRelationType") int socialRelationType, @Param(name = "serviceContext", className = "") JSONObject serviceContext);

  @Path("/delete-microblogs-entry")
  Call<JSONObject> deleteMicroblogsEntry(@Param(name = "microblogsEntryId") long microblogsEntryId);

  @Path("/get-microblogs-entries")
  Call<JSONArray> getMicroblogsEntries(@Param(name = "start") int start, @Param(name = "end") int end);

  @Path("/get-microblogs-entries")
  Call<JSONArray> getMicroblogsEntries(@Param(name = "assetTagName") String assetTagName, @Param(name = "start") int start, @Param(name = "end") int end);

  @Path("/get-microblogs-entries-count")
  Call<Integer> getMicroblogsEntriesCount();

  @Path("/get-microblogs-entries-count")
  Call<Integer> getMicroblogsEntriesCount(@Param(name = "assetTagName") String assetTagName);

  @Path("/get-microblogs-entry")
  Call<JSONObject> getMicroblogsEntry(@Param(name = "microblogsEntryId") long microblogsEntryId);

  @Path("/get-user-microblogs-entries")
  Call<JSONArray> getUserMicroblogsEntries(@Param(name = "microblogsEntryUserId") long microblogsEntryUserId, @Param(name = "start") int start, @Param(name = "end") int end);

  @Path("/get-user-microblogs-entries")
  Call<JSONArray> getUserMicroblogsEntries(@Param(name = "microblogsEntryUserId") long microblogsEntryUserId, @Param(name = "type") int type, @Param(name = "start") int start, @Param(name = "end") int end);

  @Path("/get-user-microblogs-entries-count")
  Call<Integer> getUserMicroblogsEntriesCount(@Param(name = "microblogsEntryUserId") long microblogsEntryUserId);

  @Path("/get-user-microblogs-entries-count")
  Call<Integer> getUserMicroblogsEntriesCount(@Param(name = "microblogsEntryUserId") long microblogsEntryUserId, @Param(name = "type") int type);

  @Path("/update-microblogs-entry")
  Call<JSONObject> updateMicroblogsEntry(@Param(name = "microblogsEntryId") long microblogsEntryId, @Param(name = "content") String content, @Param(name = "socialRelationType") int socialRelationType, @Param(name = "serviceContext", className = "") JSONObject serviceContext);
}

File: windows/Liferay.SDK/Service/V72/Microblogsentry/MicroblogsentryService.cs
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;

namespace Liferay.SDK.Service.V72.Microblogsentry
{
	public class MicroblogsentryService : ServiceBase
	{
		public MicroblogsentryService(ISession session)
			: base(session)
		{
		}

		public async Task<dynamic> DeleteMicroblogsEntryAsync(long microblogsEntryId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("microblogsEntryId", microblogsEntryId);

			var _command = new JsonObject()
			{
				{ "/microblogs.microblogsentry/delete-microblogs-entry", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> AddMicroblogsEntryAsync(long userId, string content, int type, long parentMicroblogsEntryId, int socialRelationType, JsonObjectWrapper serviceContext)
		{
			var _parameters = new JsonObject();

			_parameters.Add("userId", userId);
			_parameters.Add("content", content);
			_parameters.Add("type", type);
			_parameters.Add("parentMicroblogsEntryId", parentMicroblogsEntryId);
			_parameters.Add("socialRelationType", socialRelationType);
			this.MangleWrapper(_parameters, "serviceContext", "$languageUtil.getJSONWrapperClassName($parameter.type)", serviceContext);

			var _command = new JsonObject()
			{
				{ "/microblogs.microblogsentry/add-microblogs-entry", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetMicroblogsEntriesAsync(string assetTagName, int start, int end)
		{
			var _parameters = new JsonObject();

			_parameters.Add("assetTagName", assetTagName);
			_parameters.Add("start", start);
			_parameters.Add("end", end);

			var _command = new JsonObject()
			{
				{ "/microblogs.microblogsentry/get-microblogs-entries", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetMicroblogsEntriesAsync(int start, int end)
		{
			var _parameters = new JsonObject();

			_parameters.Add("start", start);
			_parameters.Add("end", end);

			var _command = new JsonObject()
			{
				{ "/microblogs.microblogsentry/get-microblogs-entries", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<long> GetMicroblogsEntriesCountAsync(string assetTagName)
		{
			var _parameters = new JsonObject();

			_parameters.Add("assetTagName", assetTagName);

			var _command = new JsonObject()
			{
				{ "/microblogs.microblogsentry/get-microblogs-entries-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<long> GetMicroblogsEntriesCountAsync()
		{
			var _parameters = new JsonObject();

			var _command = new JsonObject()
			{
				{ "/microblogs.microblogsentry/get-microblogs-entries-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<dynamic> GetMicroblogsEntryAsync(long microblogsEntryId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("microblogsEntryId", microblogsEntryId);

			var _command = new JsonObject()
			{
				{ "/microblogs.microblogsentry/get-microblogs-entry", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetUserMicroblogsEntriesAsync(long microblogsEntryUserId, int start, int end)
		{
			var _parameters = new JsonObject();

			_parameters.Add("microblogsEntryUserId", microblogsEntryUserId);
			_parameters.Add("start", start);
			_parameters.Add("end", end);

			var _command = new JsonObject()
			{
				{ "/microblogs.microblogsentry/get-user-microblogs-entries", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetUserMicroblogsEntriesAsync(long microblogsEntryUserId, int type, int start, int end)
		{
			var _parameters = new JsonObject();

			_parameters.Add("microblogsEntryUserId", microblogsEntryUserId);
			_parameters.Add("type", type);
			_parameters.Add("start", start);
			_parameters.Add("end", end);

			var _command = new JsonObject()
			{
				{ "/microblogs.microblogsentry/get-user-microblogs-entries", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<long> GetUserMicroblogsEntriesCountAsync(long microblogsEntryUserId, int type)
		{
			var _parameters = new JsonObject();

			_parameters.Add("microblogsEntryUserId", microblogsEntryUserId);
			_parameters.Add("type", type);

			var _command = new JsonObject()
			{
				{ "/microblogs.microblogsentry/get-user-microblogs-entries-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<long> GetUserMicroblogsEntriesCountAsync(long microblogsEntryUserId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("microblogsEntryUserId", microblogsEntryUserId);

			var _command = new JsonObject()
			{
				{ "/microblogs.microblogsentry/get-user-microblogs-entries-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<dynamic> UpdateMicroblogsEntryAsync(long microblogsEntryId, string content, int socialRelationType, JsonObjectWrapper serviceContext)
		{
			var _parameters = new JsonObject();

			_parameters.Add("microblogsEntryId", microblogsEntryId);
			_parameters.Add("content", content);
			_parameters.Add("socialRelationType", socialRelationType);
			this.MangleWrapper(_parameters, "serviceContext", "$languageUtil.getJSONWrapperClassName($parameter.type)", serviceContext);

			var _command = new JsonObject()
			{
				{ "/microblogs.microblogsentry/update-microblogs-entry", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}
	}
}